core/src/test/scala/unit/kafka/log/LogSegmentTest.scala

```diff
@@ -65,6 +66,21 @@ class LogSegmentTest {
     Utils.delete(logDir)
   }
 
+  /**
+   * If the maximum offset beyond index, appended to the log section, it throws LogSegmentOffsetOverflowException
+   */
+  @Test
+  def testAppendForLogSegmentOffsetOverflowException(): Unit = {
+    val seg = createSegment(0)
```

Comment:
May I suggest to create a segment with non-zero index so that we can test for https://github.com/apache/kafka/blob/trunk/storage/src/main/java/org/apache/kafka/storage/internals/log/AbstractIndex.java#L554 condition to be false. The `relativeOffset` should fit within an integer but the current test case only tests for situation where we have 0 as starting index.

Reply:
Thanks, good advice, I've modified it along with this scheme